sdk/options.go

```diff
@@ -11,7 +11,8 @@ type Option func(*config)
 // Internal config struct for building SDK options.
 type config struct {
 	tls               grpc.DialOption
-	clientCredentials oauth.ClientCredentials
+	clientCredentials *oauth.ClientCredentials
+	tokenExchange     *oauth.TokenExchangeInfo
```

Comment:
Could we create a generic interface here for the different flows?

Reply:
I like the idea but I think that if we wanted to we'd need to do something like
```
`config` -> `flow interface` -> `use by sdk`
```
because of how the functional options work; they each set an individual property that influences
how things work together.

Reply:
I also think that the `TokenSource` might be this interface. The flow is then:
1. set options on config using function options
2. turn the options into an instance of `TokenSource`
3. use this `TokenSource` whenever we want an access token